### PR TITLE
Add Sweethearts

### PR DESCRIPTION
**Pull request by BluePsychoRanger**
Sweethearts updated for 1.16

- Same version used in GM4 1.16 UHC

### CODE REVIEW

**Comment by SandSander** [comment]
Still need testing on a server?

File: gm4_sweethearts/data/gm4_sweethearts/predicates/holding_flower.json [hunk 1]
```diff
@@ -0,0 +1,137 @@
+{
+  "condition": "minecraft:alternative",
+  "terms": [
+    {
+      "condition": "minecraft:entity_properties",
+      "entity": "this",
+      "predicate": {
+        "equipment": {
+          "mainhand": {
+            "item": "minecraft:dandelion"
```
Comment (misode) [inline gm4_sweethearts/data/gm4_sweethearts/predicates/holding_flower.json:10]: Use an item tag for this

File: gm4_sweethearts/data/gm4_sweethearts/functions/main.mcfunction [hunk 2]
```diff
@@ -0,0 +1,5 @@
+execute as @a[gamemode=!creative,gamemode=!spectator,scores={gm4_sh_sneak=1..},predicate=gm4_sweethearts:holding_flower] at @s run function gm4_sweethearts:sneaking
+scoreboard players reset @a gm4_sh_sneak
```
Comment (misode) [inline gm4_sweethearts/data/gm4_sweethearts/functions/main.mcfunction:2]: You can also detect sneaking in the predicate, so you probably don't need this scoreboard objective.

**Comment by BluePsychoRanger** [comment]
Predicate optimizations have been made and this module is now ready for (re)testing and/or merging.

**Review by misode — APPROVED** [review]

**Pull request merged**